[nginx-ingress] refactor proxy-failover-iptables app

### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.46

### Expected Behavior

no restarts when 127.0.0.1:10254/healthz is unavailable
no flooding in ingress-failover iptables chain

### Actual Behavior

* The app shouldn't panic in case 127.0.0.1:10254/healthz handle is unavailable (http.resp == nil)
* The app's working logic should be reconsidered - it keeps adding duplicate entries to the host iptables on restart
```
-A ingress-failover -m socket --nowildcard -j RETURN
-A ingress-failover -p tcp -j CONNMARK --set-xmark 0x3/0xffffffff
-A ingress-failover -p tcp -j CONNMARK --set-xmark 0x3/0xffffffff
-A ingress-failover -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
-A ingress-failover -p tcp -j DNAT --to-destination 169.254.20.11:81
-A ingress-failover -p tcp -j DNAT --to-destination 169.254.20.11:444
-A ingress-failover -m socket --nowildcard -j RETURN
-A ingress-failover -p tcp -j CONNMARK --set-xmark 0x3/0xffffffff
-A ingress-failover -p tcp -j CONNMARK --set-xmark 0x3/0xffffffff
-A ingress-failover -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
-A ingress-failover -p tcp -j DNAT --to-destination 169.254.20.11:81
-A ingress-failover -p tcp -j DNAT --to-destination 169.254.20.11:444
-A ingress-failover -m socket --nowildcard -j RETURN
-A ingress-failover -p tcp -j CONNMARK --set-xmark 0x3/0xffffffff
-A ingress-failover -p tcp -j CONNMARK --set-xmark 0x3/0xffffffff
-A ingress-failover -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
-A ingress-failover -p tcp -j DNAT --to-destination 169.254.20.11:81
-A ingress-failover -p tcp -j DNAT --to-destination 169.254.20.11:444
```

### Steps To Reproduce

1. use HostWithFailover inlet for an nginx ingress controller
2. somehow prevent main controller pod from starting
3. check restarts in the proxy-failover pod and check the host's iptables (ingress-failover chain) after a few restarts

### Additional Information

_No response_

### Logs

```shell
root@dev-master-0:~# kubectl  logs -n d8-ingress-nginx proxy-main-failover-xfgkc iptables-loop -p
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x669065]

goroutine 1 [running]:
main.loop(0xc00019ff48?)
	/src/main.go:136 +0x65
main.main()
	/src/main.go:126 +0x7af
```
```

--- a/modules/402-ingress-nginx/images/proxy-failover-iptables/main.go
+++ b/modules/402-ingress-nginx/images/proxy-failover-iptables/main.go
@@ -22,6 +22,7 @@ import (
 	"net"
 	"net/http"
 	"strings"
+	"syscall"
 	"time"
 
 	"github.com/coreos/go-iptables/iptables"
@@ -84,27 +85,27 @@ func main() {
 			log.Fatal(err)
 		}
 	}
-	err = iptablesMgr.Insert("nat", chainName, 1, socketExistsRule...)
+	err = insertUnique(iptablesMgr, "nat", chainName, socketExistsRule, 1)
 	if err != nil {
 		log.Fatal(err)
 	}
-	err = iptablesMgr.Insert("nat", chainName, 2, markHttpRule...)
+	err = insertUnique(iptablesMgr, "nat", chainName, markHttpRule, 2)
 	if err != nil {
 		log.Fatal(err)
 	}
-	err = iptablesMgr.Insert("nat", chainName, 3, markHttpsRule...)
+	err = insertUnique(iptablesMgr, "nat", chainName, markHttpsRule, 3)
 	if err != nil {
 		log.Fatal(err)
 	}
-	err = iptablesMgr.Insert("nat", chainName, 4, saveMarkRule...)
+	err = insertUnique(iptablesMgr, "nat", chainName, saveMarkRule, 4)
 	if err != nil {
 		log.Fatal(err)
 	}
-	err = iptablesMgr.Insert("nat", chainName, 5, dnatHttpRule...)
+	err = insertUnique(iptablesMgr, "nat", chainName, dnatHttpRule, 5)
 	if err != nil {
 		log.Fatal(err)
 	}
-	err = iptablesMgr.Insert("nat", chainName, 6, dnatHttpsRule...)
+	err = insertUnique(iptablesMgr, "nat", chainName, dnatHttpsRule, 6)
 	if err != nil {
 		log.Fatal(err)
 	}
@@ -124,7 +125,7 @@ func main() {
 			return
 		case <-ticker.C:
 			err := loop(iptablesMgr)
-			if err != nil {
+			if err != nil && !errors.Is(err, syscall.ECONNREFUSED) {
 				log.Fatal(err)
 			}
 		}
@@ -133,16 +134,22 @@ func main() {
 
 func loop(iptablesMgr *iptables.IPTables) error {
 	resp, err := http.Get("http://127.0.0.1:10254/healthz")
-	defer resp.Body.Close()
+
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+
 	if err != nil {
-		return err
+		log.Println(err)
+		return iptablesMgr.DeleteIfExists("nat", chainName, socketExistsRule...)
 	}
 
-	if resp.StatusCode == http.StatusOK {
-		return insertUnique(iptablesMgr, "nat", chainName, socketExistsRule, 1)
+	if resp.StatusCode != http.StatusOK {
+		log.Printf("Got %d status code", resp.StatusCode)
+		return iptablesMgr.DeleteIfExists("nat", chainName, socketExistsRule...)
 	}
 
-	return iptablesMgr.DeleteIfExists("nat", chainName, socketExistsRule...)
+	return insertUnique(iptablesMgr, "nat", chainName, socketExistsRule, 1)
 }
 
 func insertUnique(iptablesMgr *iptables.IPTables, table, chain string, rule []string, pos int) error {

--- a/modules/402-ingress-nginx/images/proxy-failover-iptables/main.go
+++ b/modules/402-ingress-nginx/images/proxy-failover-iptables/main.go
@@ -135,14 +135,12 @@ func main() {
 func loop(iptablesMgr *iptables.IPTables) error {
 	resp, err := http.Get("http://127.0.0.1:10254/healthz")
 
-	if resp != nil {
-		defer resp.Body.Close()
-	}
-
 	if err != nil {
 		log.Println(err)
 		return iptablesMgr.DeleteIfExists("nat", chainName, socketExistsRule...)
 	}
+
+	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
 		log.Printf("Got %d status code", resp.StatusCode)

